storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -1351,6 +1351,63 @@ void test_write_read_erase_raw(void)
 }
 
 
+/* Erase whole flash, write test pattern on block zero and check if it appeared in any other block */
+void test_memory_aliasing(void)
+{
+	void *buffer = mmap(NULL, 16 * _PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_UNCACHED | MAP_PHYSMEM | MAP_ANONYMOUS, -1, 0x900000);
+	flashdrv_dma_t *dma;
+	int err;
+	size_t pagesPerBlock = (flashinfo.erasesz / flashinfo.writesz);
+
+	memset(buffer, 0, 16 * _PAGE_SIZE);
+
+	for (int i = 0; i < 0x1000; ++i) {
+		((char *)buffer)[i] = 0xb2;
+		((char *)buffer)[0x1000 + i] = 0x8a;
+	}
+
+	printf("test_memory_aliasing: creating\n");
+
+	dma = flashdrv_dmanew();
+
+	printf("test_memory_aliasing: chip erase\n");
+	for (size_t i = 0; i < flashinfo.size / flashinfo.erasesz; ++i) {
+		flashdrv_erase(dma, i * flashinfo.erasesz / flashinfo.writesz);
```

Comment:
- don't try to erase badblocks (we don't want to accidentally remove badblock marker - should fail, but let's not depend only on NAND logic)

- also for factory tests: all blocks are erased after `test_write_read_erase` test, so maybe make erasing optional? (to be disabled in factory?)

Reply:
Done, I'd rather not depend on what tests are enabled or they order